tests/test_plugins_meta.py

```diff
@@ -59,6 +59,13 @@ def test_unknown_plugin_has_tests(self):
                 self.assertIn(pname, self.session.plugins.keys(),
                               "{0} is not a plugin but has tests".format(pname))
 
+    def test_plugin_not_in_removed_list(self):
+        from streamlink import plugins as streamlinkplugins
+        with open(os.path.abspath(os.path.join(streamlinkplugins.__path__[0], ".removed"))) as file:
+            plugins = {name: True for name in file.read().split("\n") if name and not name.startswith("#")}
+            for pname in self.session.plugins.keys():
+                self.assertNotIn(pname, plugins, f"{pname} is not in removed plugins list")
```

Comment:
Less of a request for change, and more of a comment on language feature availability. 
You could probably use a `set` instead of a `dict` with `True` values. 

```suggestion
        with open(os.path.abspath(os.path.join(streamlinkplugins.__path__[0], ".removed"))) as file:
            plugins = {name for name in file.read().split("\n") if name and not name.startswith("#")}
            for pname in self.session.plugins.keys():
                self.assertNotIn(pname, plugins, f"{pname} is not in removed plugins list")
```